### PR TITLE
Created new operation: Instance Set Metadata

### PR DESCRIPTION
**Pull request by Tirla-Alin**
Created new operation: Instance Set Metadata
Will need unit tests.

Signed-off-by: TirlaAlin <alin.tirla@hpe.com>

### CODE REVIEW

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+/**
+  * Created by Tirla Alin
+  * 2/27/2017.
+  */
+class InstanceSetMetadata {
+
+  @Action(name = "Set Instance Metadata",
+    outputs = Array(
+      new Output(OutputNames.RETURN_CODE),
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: import RETURN_CODE ( be consistent with ZONE_OPERATION_NAME)

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+      val proxyPortStr = defaultIfEmpty(proxyPortInp, DEFAULT_PROXY_PORT)
+      val proxyPassword = defaultIfEmpty(proxyPasswordInp, DEFAULT_PROXY_PASSWORD)
+      val prettyPrint = defaultIfEmpty(prettyPrintInp, DEFAULT_PRETTY_PRINT)
+
+      val validationStream = validateProxyPort(proxyPortStr) ++
+        validatePairedLists(itemsKeysList, itemsValuesList, delimiter)
+
+      if (validationStream.nonEmpty) {
+        return getFailureResultsMap(validationStream.mkString(NEW_LINE))
+      }
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: all of the above can be out of the try block

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+      val jsonFactory = JsonFactoryUtils.getDefaultJacksonFactory
+
+      val credential = GoogleAuth.fromAccessToken(accessToken)
+
+      val result = InstanceService.setMetadata(httpTransport, jsonFactory, credential, projectId, zone, instanceName, items)
+      val resultString = if (toBoolean(prettyPrint)) result.toPrettyString else result.toString
+
+      val mapResult = getSuccessResultsMap(resultString)
+
+      mapResult.put(ZONE_OPERATION_NAME, result.getName)
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: look at ZoneOperationsGet at how to add a new entry to the map

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/InputValidator.scala [hunk 4]
```diff
@@ -30,4 +33,14 @@ object InputValidator {
   def validateNonNegativeInteger: (String, String) => Stream[String] = validate(_, _) { value =>
     if (!NumberUtilities.isValidInt(value, 0, Int.MaxValue)) Some(INVALID_NON_NEGATIVE_INTEGER) else None
   }
+
+  def validatePairedLists(list1: String, list2: String, delimiter: String): String = {
+    val firstLength = StringUtils.split(list1, delimiter).length
```
Comment (victorursan) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/InputValidator.scala]: use the split from scala "some, string".split(", ") ...

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala (hunk 2 above)
Comment (Tirla-Alin) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: I like this way, for any eventuality.

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/InputValidator.scala (hunk 4 above)
Comment (Tirla-Alin) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/InputValidator.scala]: I like the one from apache, do we really want to use the one from scala?

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala (hunk 1 above)
Comment (Tirla-Alin) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: Done

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala (hunk 3 above)
Comment (Tirla-Alin) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/actions/compute/instances/InstanceSetMetadata.scala]: Done

File: cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/OutputNames.scala [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package io.cloudslang.content.gcloud.utils.action
+
+/**
+  * Created by Tirla Alin on 28.02.2017.
+  */
+object OutputNames {
+  final val ZONE_OPERATION_NAME = "zoneOperationName"
```
Comment (robert-sandor) [inline cs-google-cloud/src/main/scala/io/cloudslang/content/gcloud/utils/action/OutputNames.scala]: you can use the one from GoogleOutputNames

**Comment by Tirla-Alin** [comment]
@victorursan can this PR be merged?

**Pull request merged**